code-comment-style.md

```diff
@@ -0,0 +1,128 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- File
+- Type
+- Constant
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment
+
+    - Used for inline code comment (for better code explaination when reviewing)
+    - Use **//** for a line comment
+    - Use **/* … */** for a block comment (used only when needed)
+    
+- Doc comment
+
+    - Used in module level descriptions (structures, fields, macros, etc.) for command line reference
+    - Use **///** for a line comment 
+    - Use **//!** for a block comment
+    - Use **rustfmt** to format your code
+
+- Place the single-line and block comment above the code it’s annotating
+- Fold long lines of comments
+- Each line of text in your code and comment should be at most 100 characters long
+- Do not use relative URLs; use **absolute URLs** instead
+
+    Right:
+    
+    ```
+    // For more configuration details, see https://example.com/project/tools/config-items.
+    ```
+    
+    Wrong:
+    
+    ```
+    // For more details, see [Configuration Items](../tools/config-items.md).
+    ```
+
+### Language for a good comment
+
+- Word
+    
+    - Use **American English** rather than British English
+        
+        - color, canceling, synchronize     (Recommended)
+        - colour, cancelling, synchronise   (Not recommended)
+    
+    - Use correct spelling
+
+    - Use **standard or official capitalization**
+        
+        - TiKV, TiDB-Binlog, Region, gRPC, RocksDB, GC, k8s, [mydumper](https://github.com/maxbube/mydumper), [Prometheus Pushgateway](https://github.com/prometheus/pushgateway)   (Right)
+        - Tikv, TiDB Binlog, region, grpc, rocksdb, gc, k8S, MyDumper, Prometheus PushGateway   (Wrong)
+
+    - Use words and expressions consistently
+        
+        - dead link vs. broken link (Only one of them can appear in a single document)
+    
+    - Do not use lengthy compound words
+
+    - Do not abbreviate unless it is absolutely necessary

```

Comment:
I think abreviations can be fine if they improve readability, but avoid overusing them. For example, it would be tedious to read 'Titanium Data Base' all the time instead of 'TiDB', or to spell out 'HTTP'

Reply:
True. Since this is code comment we do not have to follow this rule as strictly as documentation. For general accepted/known abbreviations, we can and should use to improve readability. For less-known items, we may still need to spell out